### PR TITLE
Add Unit Test

### PR DESCRIPTION
**Pull request by sivasuriyankumarasamy**

### CODE REVIEW

**Comment by sivasuriyankumarasamy** [comment]
Hi @egonw - I have added unit test case for below mentioned file. 

- CMLDictionary.java
- CMLGradient
- CMLIdentifier
- CMLIsotope

Kindly review and provide feedback if any.

File: src/test/java/org/xmlcml/cml/element/main/CMLDictionaryTest.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.xmlcml.cml.element.CMLDictionary;
+import org.xmlcml.cml.element.CMLEntry;
+
+public class CMLDictionaryTest {
+
+    protected CMLDictionary cmlDictionary = null;
```
Comment (egonw) [inline src/test/java/org/xmlcml/cml/element/main/CMLDictionaryTest.java]: Depending on how to JUnit tests are run, the individual tests are not necessarily run in the order in which they are given in the source code, particularly, when you run tests in parallel. I have had problems with this in the past. It makes the tests slower, I know, but may I suggest to have a new `CMLDictionary` with each test, and not use global variables? I totally understand you did, but it's more stable without.

File: src/test/java/org/xmlcml/cml/element/main/CMLIdentifierTest.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
+/**
+ *    Copyright 22023 Sivasuriyan
```
Comment (egonw) [inline src/test/java/org/xmlcml/cml/element/main/CMLIdentifierTest.java]: Plz update the year :)

**Review by egonw — CHANGES_REQUESTED** [review]

File: src/test/java/org/xmlcml/cml/element/main/CMLDictionaryTest.java (hunk 1 above)
Comment (sivasuriyankumarasamy) [reply src/test/java/org/xmlcml/cml/element/main/CMLDictionaryTest.java]: Done

File: src/test/java/org/xmlcml/cml/element/main/CMLIdentifierTest.java (hunk 2 above)
Comment (sivasuriyankumarasamy) [reply src/test/java/org/xmlcml/cml/element/main/CMLIdentifierTest.java]: Done

**Comment by sivasuriyankumarasamy** [comment]
Hi @egonw,

Added the requested changes.

**Review by egonw — APPROVED** [review]
Thank you for the updates

**Comment by egonw** [comment]
Merged into the 4.7 dev branch: https://github.com/BlueObelisk/cmlxom/pull/22

**Pull request closed without merging**